### PR TITLE
Log start of cluster sync

### PR DESCRIPTION
**Pull request by borancar**
It's helpful to operators to understand at which commit the cluster is
stuck should any errors occur with cluster sync, such as k8s internal
issues:
- https://github.com/fluxcd/flux/issues/1991
- https://github.com/jetstack/cert-manager/issues/2273

### CODE REVIEW

**Comment by borancar** [comment]
To give some more context, I spent time looking at all the wrong places why the cluster wasn't getting updated until I finally got Delve and narrowed down the issue. The following 2 messages were actually not very helpful in understanding why the images were stuck at their old version:
```
ts=2020-02-04T20:29:52.312079303Z caller=main.go:267 type="internal kubernetes error" kubernetes_caller=k8s.io/client-go@v11.0.0+incompatible/discovery/cached/memory/memcache.go:199 err="couldn't get resource list for admission.certmanager.k8s.io/v1beta1: the server is currently unable to handle the request"
ts=2020-02-04T20:29:52.329078854Z caller=main.go:267 type="internal kubernetes error" kubernetes_caller=k8s.io/client-go@v11.0.0+incompatible/discovery/cached/memory/memcache.go:111 err="couldn't get resource list for admission.certmanager.k8s.io/v1beta1: the server is currently unable to handle the request"
```
Having those messages after a message saying "Trying/Attempting sync..." with specific git versions would have helped me massively narrow down on the issue.

File: pkg/daemon/sync.go [hunk 1]
```diff
@@ -70,6 +70,8 @@ func (d *Daemon) Sync(ctx context.Context, started time.Time, newRevision string
 		return err
 	}
 
+	d.Logger.Log(fmt.Sprintf("Trying to sync git changes to cluster, from commit %s to commit %s", c.oldTagRev, c.newTagRev))
```
Comment (hiddeco) [inline pkg/daemon/sync.go]:
```suggestion
	d.Logger.Log(fmt.Sprintf("Trying to sync git changes to cluster, from commit '%s' to commit '%s'", c.oldTagRev, c.newTagRev))
```

Would be easier to read in a blob of logs imo.

File: pkg/daemon/sync.go [hunk 2]
```diff
@@ -70,6 +70,8 @@ func (d *Daemon) Sync(ctx context.Context, started time.Time, newRevision string
 		return err
 	}
 
+	d.Logger.Log(fmt.Sprintf("Trying to sync git changes to cluster, from commit '%s' to commit '%s'", c.oldTagRev, c.newTagRev))
```
Comment (hiddeco) [inline pkg/daemon/sync.go]: I missed the fact that you did not include a key for the message, I think you have two options:

```suggestion
	d.Logger.Log("info", fmt.Sprintf("trying to sync git changes to the cluster, from commit '%s' to commit '%s'", c.oldTagRev, c.newTagRev))
```

or

```suggestion
	d.Logger.Log("info", "trying to sync git changes to the cluster", "old", c.oldTagRev, "new", c.neTagRev)
```

The latter matches https://github.com/fluxcd/flux/blob/4ed1b29667bbb0a82872c7201ce965016e3c68a7/pkg/daemon/loop.go#L220

Comment (borancar) [reply pkg/daemon/sync.go]: Thanks, the latter makes more sense. Just need to fix a typo (c.newTagRev vs c.neTagRev) and pushing.

Comment (hiddeco) [reply pkg/daemon/sync.go]: Can you please smash your commits together and rebase on `master` too?

**Comment by hiddeco** [comment]
Can you please smash your commits together, edit the commit message and remove the issue links from it (to prevent endless pings to the issues from our friendly OSS neighbours), and rebase the result on `master`?

**Pull request merged**